src/expiryWarning.ts

```diff
@@ -0,0 +1,84 @@
+import moment from 'moment';
+import { promisify } from 'util';
+import { resolve, sep } from 'path';
+import fs from 'fs';
+const readdir = promisify(fs.readdir);
+const stat = promisify(fs.stat);
+
+import {
+  DEFAULT_RECORDING_DIR,
+  DEFAULT_EXPIRATION_DAYS,
+  DEFAULT_DAYS_FOR_EXPIRY_WARNING,
+  COLORS,
+} from './constants';
+
+// TYPES
+interface recordingFile {
+  name: string,
+  status: string,
+}
+
+// CONSTANTS
+const DAYS_TO_WARN: number = process.env.POLLY_DAYS_TO_WARN ? parseInt(process.env.POLLY_DAYS_TO_WARN, 10) : DEFAULT_DAYS_FOR_EXPIRY_WARNING;
+const DAYS_EXPIRY: number = process.env.POLLY_DAYS_EXPIRY ? parseInt(process.env.POLLY_DAYS_EXPIRY, 10) : DEFAULT_EXPIRATION_DAYS;
+const RECORDINGS_DIR: string = process.env.POLLY_RECORDINGS_DIR || DEFAULT_RECORDING_DIR;
+const STATES = {
+  EXPIRED: `${COLORS.RED}Expired${COLORS.DEFAULT}`,
+  ALMOST_EXPIRED: `${COLORS.YELLOW}Expires in less than ${DEFAULT_DAYS_FOR_EXPIRY_WARNING} days${COLORS.DEFAULT}`,
+};
+
+// Get all expired or soon to be expired recordings
+export async function getFiles(dir: string, fileList: recordingFile[] = []) {
+  let recordingsDirContent;
+  try {
+    recordingsDirContent = await readdir(dir);
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.error(`[Polly Jest Preset] Recordings directory "${dir}" not found`);
```

Comment:
feels weird to have a full try-catch to test for file existence. any reason to not use `fs.stat()` instead?

Reply:
also weird to do this because this is recursive, so could fail on leaf nodes